tachyon/base/buffer/read_only_buffer.h

```diff
@@ -133,6 +134,22 @@ class TACHYON_EXPORT ReadOnlyBuffer {
     return true;
   }
 
+  template <typename T>
+  [[nodiscard]] bool ReadPtrAt(size_t buffer_offset, T** ptr,
+                               size_t ptr_num) const {
+    size_t size = sizeof(T) * ptr_num;
+    base::CheckedNumeric<size_t> len = buffer_offset;
+    size_t size_needed;
+    if (!(len + size).AssignIfValid(&size_needed)) return false;
+    if (size_needed > buffer_len_) {
+      return false;
+    }
```

Comment:
```suggestion
    if (size_needed > buffer_len_) return false;
```
This is optional and you may resolve this comment if you think it's not necessary.

Reply:
I agreed that this looks better, but if you see `tachyon/baes/buffer/read_only_buffer.cc`, there are 2 more places that use this pattern, so I'll resolve Thanks :)